src/CryptorWorker.ts

```diff
@@ -0,0 +1,28 @@
+import { expose } from "threads"
+import * as crypto from 'crypto'
+
+let _cipher: crypto.Cipher
+let _decipher: crypto.Decipher
+
+const _cryptorWorker = {
+  init(algorithm: string, key: Buffer, initVector: Buffer): void {
+		_cipher = crypto.createCipheriv(algorithm, key, initVector)
+		_decipher = crypto.createDecipheriv(algorithm, key, initVector)
+  },
+  updateCipher(algorithm: string, key: ArrayBuffer, initVector: ArrayBuffer, plainBuf: crypto.BinaryLike): Buffer {
+    return _cipher.update(plainBuf)
+  },
+  _resetCipher(algorithm: string, key: crypto.CipherKey, initVector: crypto.BinaryLike): void {
```

Comment:
What exactly is `_resetCipher` used for?

Reply:
it is used to reinitialise the cipher with a new `initVector`